### PR TITLE
Feature: data streams

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/CreateDataStream/interface.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { RouteComponentProps } from "react-router-dom";
```
Comment (Hailong-am) [inline public/pages/CreateDataStream/interface.ts:1]: copyright is missing

File: public/pages/CreateDataStream/hooks.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { EuiFormRowProps } from "@elastic/eui";
```
Comment (Hailong-am) [inline public/pages/CreateDataStream/hooks.tsx:1]: copyright missing

File: public/pages/CreateDataStream/containers/TemplateMappings/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import TemplateMappings from "./TemplateMappings";
```
Comment (Hailong-am) [inline public/pages/CreateDataStream/containers/TemplateMappings/index.ts:1]: copyright missing

File: public/pages/CreateDataStream/containers/TemplateMappings/TemplateMappings.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React, { useContext, useRef } from "react";
```
Comment (Hailong-am) [inline public/pages/CreateDataStream/containers/TemplateMappings/TemplateMappings.tsx:1]: copyright missing

File: public/pages/CreateDataStream/containers/IndexAlias/index.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import IndexAlias from "./IndexAlias";
```
Comment (Hailong-am) [inline public/pages/CreateDataStream/containers/IndexAlias/index.ts:1]: copyright missing

File: public/pages/CreateDataStream/containers/IndexAlias/IndexAlias.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
+import React, { useContext } from "react";
```
Comment (Hailong-am) [inline public/pages/CreateDataStream/containers/IndexAlias/IndexAlias.tsx:1]: copyright missing

File: public/pages/CreateDataStream/containers/DefineDataStream/index.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+import DefineDataStream from "./DefineDataStream";
```
Comment (Hailong-am) [inline public/pages/CreateDataStream/containers/DefineDataStream/index.ts:1]: copyright missing

File: public/pages/CreateDataStream/containers/DefineDataStream/DefineDataStream.tsx [hunk 8]
```diff
@@ -0,0 +1,224 @@
+import React, { useRef, useMemo } from "react";
```
Comment (Hailong-am) [inline public/pages/CreateDataStream/containers/DefineDataStream/DefineDataStream.tsx:1]: copyright missing

File: public/pages/CreateDataStream/containers/DataStreamDetail/hooks.tsx [hunk 9]
```diff
@@ -0,0 +1,71 @@
+import { CoreStart } from "opensearch-dashboards/public";
```
Comment (Hailong-am) [inline public/pages/CreateDataStream/containers/DataStreamDetail/hooks.tsx:1]: copyright missing

File: public/pages/CreateDataStream/containers/BackingIndices/index.ts [hunk 10]
```diff
@@ -0,0 +1,3 @@
+import BackingIndices from "./BackingIndices";
```
Comment (Hailong-am) [inline public/pages/CreateDataStream/containers/BackingIndices/index.ts:1]: copyright missing

File: public/pages/CreateDataStream/containers/BackingIndices/BackingIndices.tsx [hunk 11]
```diff
@@ -0,0 +1,181 @@
+import React, { useContext, useEffect, useState } from "react";
```
Comment (Hailong-am) [inline public/pages/CreateDataStream/containers/BackingIndices/BackingIndices.tsx:1]: copyright missing

File: public/pages/CreateDataStream/components/IndexSettings/index.ts [hunk 12]
```diff
@@ -0,0 +1,3 @@
+import IndexSettings from "./IndexSettings";
```
Comment (Hailong-am) [inline public/pages/CreateDataStream/components/IndexSettings/index.ts:1]: copyright missing

File: public/pages/CreateDataStream/components/IndexSettings/IndexSettings.tsx [hunk 13]
```diff
@@ -0,0 +1,122 @@
+import React from "react";
```
Comment (Hailong-am) [inline public/pages/CreateDataStream/components/IndexSettings/IndexSettings.tsx:1]: copyright missing

**Review by xluo-aws — DISMISSED** [review]

**Comment by SuZhou-Joe** [comment]
@Hailong-am For copyright, will add that in another branch after all the dev branches are merged.

**Comment by Angie-Zhang** [comment]
Replaced label v2.6.0 with v2.x due to the 2.6.0 release code cut off.

**Review by Hailong-am — APPROVED** [review]

**Review by Hailong-am — DISMISSED** [review]

**Review by xluo-aws — DISMISSED** [review]

**Review by xluo-aws — DISMISSED** [review]

**Review by Hailong-am — APPROVED** [review]

**Review by Hailong-am — APPROVED** [review]

**Review by xluo-aws — APPROVED** [review]

**Pull request merged**